File: src/main/java/me/sul/servercore/datasaveschedule/DataSaveScheduler.java
package me.sul.servercore.datasaveschedule;

import me.sul.servercore.ServerCore;
import org.bukkit.event.Event;
import org.bukkit.event.Listener;
import org.bukkit.plugin.Plugin;
import org.bukkit.scheduler.BukkitRunnable;

import java.util.logging.Level;

public class DataSaveScheduler implements Listener {
    private static final int SAVE_INTERVAL = 5;  // min

    public DataSaveScheduler() {
        new BukkitRunnable() {
            public void run() {
                ServerCore.getInstance().getServer().getPluginManager().callEvent((Event)new DataSaveScheduleEvent(false));
                ServerCore.getInstance().getLogger().log(Level.INFO, "서버 저장시간 : " + SAVE_INTERVAL + "m");
            }
        }.runTaskTimer((Plugin) ServerCore.getInstance(), SAVE_INTERVAL * 1200, SAVE_INTERVAL * 1200);
    }
}

File: src/main/kotlin/kr/sul/servercore/datasaveschedule/DataSaveScheduler.kt
package kr.sul.servercore.datasaveschedule

import kr.sul.servercore.ServerCore
import org.bukkit.scheduler.BukkitRunnable
import java.util.logging.Level

object DataSaveScheduler {
    private const val SAVE_INTERVAL = 5 // min

    init {
        object : BukkitRunnable() {
            override fun run() {
                ServerCore.instance.server.pluginManager.callEvent(DataSaveScheduleEvent(false))
                ServerCore.instance.logger.log(Level.INFO, "서버 저장시간 : ${SAVE_INTERVAL}m")
            }
        }.runTaskTimer(ServerCore.instance, SAVE_INTERVAL * 1200L, SAVE_INTERVAL * 1200L)
    }
}